src/octopeer-github/content/EventTracker/MouseClickTracker.ts

```diff
@@ -0,0 +1,18 @@
+/**
+ * Created by Mathias on 2016-05-27.
+ */
+class MouseClickTracker {
+
+    /**
+     * Initialize a MouseClickTracker that contains a MouseClickDatabaseAdaptable.
+     * @param db The DatabaseAdaptable for the Tracker.
+     */
+    constructor(private db: MouseClickDatabaseAdaptable) {
+        const self = this;
+        const $window = $(window);
+        $window.off("click");
+        $window.on("click", function (eventObject: JQueryEventObject) {
+            self.db.postMouseClick(new MouseClickEvent(new Date().getTime()), EMPTY_CALLBACK, EMPTY_CALLBACK);
```

Comment:
In a database meeting, we agreed on forcing a MousePositionEvent to accompany a MouseClickEvent so that the mouse_x and mouse_y wouldn't have to be sent with a click event. This does mean that the MouseClickTracker needs both a `MouseClickDatabaseAdaptable` and a `MousePositionDatabaseAdaptable`. Usually they are the same, so I'd also make an overloaded constructor like this:
```
constructor(private dbClick: MouseClickDatabaseAdaptable, private dbPosition: MousePositionDatabaseAdaptable) {
    // Do all the things you usually do
}
constructor(db: MouseClickDatabaseAdaptable & MousePositionDatabaseAdaptable) {
    this.constructor(db, db); // or however you call an overloaded constructor in typescript
}
```